app/models/patient_changeset.rb

```diff
@@ -54,6 +54,16 @@ class PatientChangeset < ApplicationRecord
           )
         end
 
+  scope :with_pds_match, -> { where.not(pds_nhs_number: nil) }
+
+  scope :with_pds_search_attempted,
+        -> do
+          where("pending_changes -> 'search_results' != '[]'::jsonb").where.not(
+            "jsonb_array_length(pending_changes -> 'search_results') = 1 AND 
+            (pending_changes -> 'search_results' -> 0 ->> 'result') = 'no_postcode'"
+          )
+        end
```

Comment:
This query has the potential to be quite slow, as it requires reading from the JSON information, I think it's okay for now but one to keep an eye on if it becomes problematic.